src/validate.ts

```diff
@@ -111,29 +112,29 @@ export function validate(intentRequest: object, apiResponse: object, syncData?:
     const executionLength = execution.length;
 
     const validateExecuteAPI = responseValidation(apiResponse, EXECUTE_RESPONSE_SCHEMA);
-   
+
     if (validateExecuteAPI) {
-        executeErrors.push(...validateExecuteAPI);
-        return executeErrors;
+      executeErrors.push(...validateExecuteAPI);
+      return executeErrors;
     }
-    
-   // identifies the part of the api response to validate against a schema
+
+    // identifies the part of the api response to validate against a schema
     const commands = apiResponse['payload']['commands'];
     const commandsLength = commands.length;
     for (let i = 0; i < executionLength; i++) {
       // gets the specific command
       const commandName = execution[i]['command'];
-      if (commandName in COMMAND_STATES_EXPECT) {
+      if (commandName in EXECUTE_COMMAND_STATES_EXPECT) {
         for (let j = 0; j < commandsLength; j++) {
-          const states = commands[j]['states'];
-          const validateExecTraitStates = responseValidation(states, COMMAND_STATES_EXPECT[commandName]);
+          const states = commands[j]['states'] || {};
```

Comment:
Maybe we should instead check that `states` exists at all and if not we can skip the validation check

Reply:
We should never skip the validation check. If the related schema requires certain fields, validation should report that they are missing since the developer provided nothing.